ci/pipelines/base-images.yml

```diff
@@ -0,0 +1,542 @@
+    repository: ((ci-registry-org.gcr))/dispatch-nodejs-base-image
+
+
+# java base image pr
+- name: java-base-image-pr
+  type: pull-request
+  source:
+    repo: dispatchframework/java-base-image
+    uri: git@github.com:dispatchframework/java-base-image
+    access_token: ((github-access-token))
+    private_key: ((github-private-key))
+    every: true
+
+- name: java-base-image
+  type: docker-image
+  source:
+    username: ((ci-registry-username.gcr))
+    password: ((ci-registry-password.gcr))
+    repository: ((ci-registry-org.gcr))/dispatch-java-base-image
+
+
+# clojure base image pr
+- name: clojure-base-image-pr
+  type: pull-request
+  source:
+    repo: dispatchframework/clojure-base-image
+    uri: git@github.com:dispatchframework/clojure-base-image
+    access_token: ((github-access-token))
+    private_key: ((github-private-key))
+    every: true
+
+- name: clojure-base-image
+  type: docker-image
+  source:
+    username: ((ci-registry-username.gcr))
+    password: ((ci-registry-password.gcr))
+    repository: ((ci-registry-org.gcr))/dispatch-clojure-base-image
+
+
+# powershell base image pr
+- name: powershell-base-image-pr
+  type: pull-request
+  source:
+    repo: dispatchframework/powershell-base-image
+    uri: git@github.com:dispatchframework/powershell-base-image
+    access_token: ((github-access-token))
+    private_key: ((github-private-key))
+    every: true
+
+- name: powershell-base-image
+  type: docker-image
+  source:
+    username: ((ci-registry-username.gcr))
+    password: ((ci-registry-password.gcr))
+    repository: ((ci-registry-org.gcr))/dispatch-powershell-base-image
+
+
+jobs:
+
+- name: create-gke-cluster
+  public: true
+  plan:
+  - get: dispatch
+    resource: dispatch-master
+  - task: create-gke-cluster
+    file: dispatch/ci/base-images/gke-cluster-create.yml
+    params:
+      GKE_KEY: ((gke-key))
+      GKE_PROJECT_ID: ((gke-project-id))
+      CLUSTER_NAME_SUFFIX: job-base-image
+  - put: k8s-cluster
+    params:
+      file: k8s-cluster/name
+
+- name: delete-gke-cluster
+  public: true
+  plan:
+  - aggregate:
+    - get: dispatch
+      resource: dispatch-master
+    - get: k8s-cluster
+      passed: [create-gke-cluster]
+  - task: delete-gke-cluster
+    file: dispatch/ci/base-images/gke-cluster-delete.yml
+    params:
+      GKE_KEY: ((gke-key))
+      GKE_PROJECT_ID: ((gke-project-id))
+
+
+- name: deploy-dispatch
+  public: true
+  plan:
+  - aggregate:
+    - get: dispatch
+      resource: dispatch-master
+    - get: dispatch-release
+      resource: dispatch-release
+      trigger: true
+    - get: k8s-cluster
+      passed: [create-gke-cluster]
+    - get: version
+  - task: deploy-dispatch
+    file: dispatch/ci/base-images/install-dispatch.yml
+    params:
+      GKE_KEY: ((gke-key))
+      GKE_PROJECT_ID: ((gke-project-id))
+      DOCKER_REGISTRY_HOST: ((docker-hub-org))
+      FAAS: openfaas
+      EVENT_TRANSPORT: rabbitmq
+    input_mapping:
+      version: version
```

Comment:
NIT: this shouldn't be needed if the name is the same, did you find it not working without the mapping?

Reply:
also since you are getting the dispatch-resource, can't you extract version from it, is the version resource needed?